src/main.ts

```diff
@@ -0,0 +1,250 @@
+  config: { cachedRepositoriesDirectoryPath, defaultProjectNames },
+}: {
+  argv: string[];
+  stdout: SimpleWriteStream;
+  stderr: SimpleWriteStream;
+  config: {
+    cachedRepositoriesDirectoryPath: string;
+    defaultProjectNames: string[];
+  };
+}) {
+  const outputLogger = new OutputLogger({ stdout, stderr });
+  const workingDirectoryPath = WORKING_DIRECTORY_PATH;
+
+  const inputs = await parseInputs({ argv, outputLogger, defaultProjectNames });
+  /* istanbul ignore next: At the moment, there is no real way that Yargs could fail */
+  if (!inputs) {
+    process.exitCode = 1;
+    return;
+  }
+  const { templateRepositoryName, projectReferences } = inputs;
+
+  const template = await establishMetaMaskRepository({
+    repositoryReference: templateRepositoryName,
+    workingDirectoryPath,
+    cachedRepositoriesDirectoryPath,
+    outputLogger,
+  });
+
+  await lintProjects({
+    projectReferences,
+    template,
+    workingDirectoryPath,
+    cachedRepositoriesDirectoryPath,
+    outputLogger,
+  });
+}
+
+/**
+ * Parses command-line arguments and establishes parameters for the remainder of
+ * this tool.
+ *
+ * @param args - The arguments to this function.
+ * @param args.argv - The name of this executable and its arguments (as obtained
+ * via `process.argv`).
+ * @param args.outputLogger - Writable streams for output messages.
+ * @param args.defaultProjectNames - The set of MetaMask repositories
+ * that will be linted.
+ * @returns The parsed command-line arguments if parsing was successful, or null
+ * otherwise.
+ */
+async function parseInputs({
+  argv,
+  outputLogger,
+  defaultProjectNames,
+}: {
+  argv: string[];
+  outputLogger: OutputLogger;
+  defaultProjectNames: string[];
+}): Promise<Inputs | null> {
+  const args = await parseCommandLineArguments({ argv, outputLogger });
+
+  if (args) {
+    const projectReferences = args._.length > 0 ? args._ : defaultProjectNames;
+    return {
+      templateRepositoryName: DEFAULT_TEMPLATE_REPOSITORY_NAME,
+      projectReferences,
+    };
+  }
+
+  /* istanbul ignore next: At the moment, there is no real way that Yargs could fail */
+  return null;
+}
+
+/**
+ * Parses the command line using `yargs`.
+ *
+ * @param args - The arguments to this function.
+ * @param args.argv - The name of this executable and its arguments (as obtained
+ * via `process.argv`).
+ * @param args.outputLogger - Writable streams for output messages.
+ */
+async function parseCommandLineArguments({
+  argv,
+  outputLogger,
+}: {
+  argv: string[];
+  outputLogger: OutputLogger;
+}) {
+  let yargsFailure: { message: string; error: Error } | null = null;
+  /* istanbul ignore next: At the moment, there is no real way that Yargs could fail */
+  const onFail = (message: string, error: Error) => {
+    if (error) {
+      throw error;
+    }
+    yargsFailure = { message, error };
+  };
+
+  const yargs = createYargs(hideBin(argv))
+    .usage(USAGE_TEXT)
+    .help(false)
+    .string('_')
```

Comment:
What necessitated this setting? I see that the docs say that this results in "non-hyphenated arguments being interpreted as strings, regardless of whether they resemble numbers." But we don't have any arguments yet, strings or otherwise

Reply:
We do have arguments. You can run this tool two ways: if you pass arguments, they're assumed to be the short names of MetaMask repos, and the tool will run the linter against just those repos; otherwise, if you don't pass anything, it will use a default set of repos.

Reply:
In any case, I added this line to see if I could get `_` typed as `string[]` instead of `(string | number)[]`, but this merely changed the type to `string & (string | number)[]`, which is strange. In any case, we don't need to do this, since we're typecasting `_` to strings below anyway, so I'll drop this.

Reply:
Removed here: [dfba17e](https://github.com/MetaMask/module-lint/pull/33/commits/dfba17e7b5f4f2ae2cb6fbd653e8cca10b1391d1)